openedx/adg/admin_dashboard/performance/views.py

```diff
@@ -1,3 +1,146 @@
+from django.db.models import Prefetch, Q
+from django.views.generic import ListView, View
+from opaque_keys.edx.keys import CourseKey
+
+from common.djangoapps.student.models import CourseEnrollment
+from lms.djangoapps.courseware.courses import get_course_by_id
+from openedx.adg.admin_dashboard.constants import DEFAULT_PAGINATE_BY
+from openedx.adg.admin_dashboard.helpers import get_http_response_with_csv_file_attachment
+from openedx.adg.admin_dashboard.performance.helpers import get_course_grades_for_users, get_course_grades_report_data
+from openedx.adg.lms.applications.models import BusinessLine, MultilingualCourseGroup
+from openedx.adg.lms.applications.rules import is_adg_admin
+
+
+class CourseGradesView(UserPassesTestMixin, ListView, View):
+    """
+    View to list and download grades of a course.
+    """
+
+    DEFAULT_COURSE_GRADES_PAGINATE_BY = DEFAULT_PAGINATE_BY
+    course = None
+    template_name = 'adg/admin_dashboard/performance/course_grades.html'
+    http_method_names = ('get', 'post')
+
+    def test_func(self):
+        """
+        Checks user permissions to access course grades.
+        Authenticated and staff user will be able to access course grades if he/she is a
+        superuser | adg admin | bu admin and course is a bu prerequisite for the same bu.
+
+        Returns:
+            bool: True if user has all permissions, False otherwise
+        """
+        user = self.request.user
+        if not (user.is_authenticated and user.is_staff):
+            return False
+
+        if user.is_superuser or is_adg_admin(user):
+            return True
+
+        if not user.groups.filter(id__in=BusinessLine.objects.values_list('group__id')).exists():
+            return False
+
+        course_key = CourseKey.from_string(self.kwargs['course_key_string'])
+        return MultilingualCourseGroup.objects.filter(
+            multilingual_courses__course__id=course_key
+        ).filter(
+            Q(is_common_business_line_prerequisite=True) | Q(business_line_prerequisite__group__in=user.groups.all())
+        ).exists()
+
+    def setup(self, request, *args, **kwargs):
+        """
+        Overridden to set course as class attribute.
+        """
+        super().setup(request, *args, **kwargs)
+        self.course = self._get_course()
+
+    def get_context_data(self, *, object_list=None, **kwargs):
+        """
+        Overridden to add course name and course grades for the users.
+
+        Returns:
+            dict: Dict containing context data.
+        """
+        object_list = self.get_queryset().only('id', 'email', 'profile__name')
+        context = super().get_context_data(object_list=object_list)
+
+        user_ids = [user.id for user in context['object_list']]
+        context['course_grades'] = get_course_grades_for_users(user_ids, self.course.id)
+        context['course_name'] = self.course.display_name
+
+        return context
+
+    def post(self, request, *args, **kwargs):
```

Comment:
Why are we using the post method to download a CSV report? The POST method is generally meant to accept user input and store it in the server, like form data, a file etc. But in this case, we are not storing anything to the server, but just sending out a report to the user. Shouldn't we be using the GET request for this?